onnxruntime/core/framework/utils.cc

```diff
@@ -633,12 +635,24 @@ void DumpNodeOutputs(OpKernelContext& context, const Node& node, const SessionSt
             if (!provider) {
               provider = cpu_execution_provider;
             }
+            std::cout << " Provider=" << provider->Type() << "\n";
 
+            const auto data_type = tensor.DataType();
             if (provider == cpu_execution_provider || tensor_location.mem_type == OrtMemTypeCPUOutput) {
-              const auto data_type = tensor.DataType();
               DispatchOnTensorType(data_type, DumpTensor, tensor, shape);
-            } else {
-              std::cout << " is not on CPU. Provider=" << provider->Type() << "\n";
+            } else if (provider == cuda_execution_provider) {
```

Comment:
>else if (provider == cuda_execution_provider) { [](start = 14, length = 47)

you may add these cuda related code inside a #ifdef USE_CUDA

Reply:
That is right. I update it to pass cpu builds.